tests/test.py

```diff
@@ -294,42 +303,123 @@ def run_terraform(*, image: str, final: bool = False):
     )
     num_tests_ran += 1
 
-    # Ensure autodetect finds the appropriate terraform configs, which can be
-    # inferred by the number of logs written to /var/log/easy_infra.log
+    # Ensure autodetect finds the appropriate terraform configs, which can be inferred by the number of logs written to /var/log/easy_infra.log
     #
-    # This test requires LEARNING_MODE to be true because the autodetect
-    # functionality traverses into the testing sub-directories, including those
-    # which are purposefully insecure, which otherwise would exit non-zero
-    # early, resulting in a limited set of logs
+    # This test requires LEARNING_MODE to be true because the autodetect functionality traverses into the testing sub-directories, including those
+    # which are purposefully insecure, which otherwise would exit non-zero early, resulting in a limited set of logs.
     test_terraform_dir = tests_test_dir.joinpath("terraform")
-    # There is always one log for each security tool, regardless of if that
-    # tool is installed in the image being used.  If a tool is not in the PATH
-    # and executable, a log message indicating that is generated
-    LOG.debug("Testing autodetect mode")
+    # There is always one log for each security tool, regardless of if that tool is installed in the image being used.  If a tool is not in the PATH
+    # and executable, a log message indicating that is generated.
     number_of_security_tools = len(CONFIG["commands"]["terraform"]["security"])
-    number_of_testing_folders = len(
-        [dir for dir in test_terraform_dir.iterdir() if dir.is_dir()]
-    )
-    autodetect_environment = copy.deepcopy(informational_environment)
+    all_test_folders = [dir for dir in test_terraform_dir.rglob("*") if dir.is_dir()]
+    test_folders_containing_only_files = []
+    for directory in all_test_folders:
+        items_in_dir = directory.iterdir()
+        for item in items_in_dir:
+            if item.is_dir():
+                break
+        else:
+            test_folders_containing_only_files.append(directory)
```

Comment:
This approach and the folder reorganization allows us to run general tests and tool-specific tests separately, which is primarily useful when we are enabling AUTODETECT.